rfcs/0000-standardize_the_kuadrant_spec.md

```diff
@@ -0,0 +1,238 @@
+# Standardize the Kuadrant Spec
+
+- Feature Name: Standardize the Kuadrant Spec
+- Start Date: 10/01/2025
+- RFC PR: [Kuadrant/architecture#0000](https://github.com/Kuadrant/architecture/pull/0000)
+- Issue tracking: [Kuadrant/architecture#0000](https://github.com/Kuadrant/architecture/issues/0000)
+
+# Summary
+[summary]: #summary
+
+This RFC is trying to describe a common API design that can be used when adding feature to Kuadrant that are configurable thought the Kuadrant CR.
+
+# Motivation
+[motivation]: #motivation
+
+As we are move forward with the development of Kuadrant, we are wanting to add new features to the product.
+Some of these features will be configurable by the user, the Kuadrant CR is an expected place for this to happen. 
+Before starting to add these configurable features, a standardize way of defining features in the Kuadrant CR should be agreed upon. 
+This RFC lays the groundwork for an API design which the new features will follow.
+By having a standardized API design, future development can move faster as the API design will have the groundwork already done.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+The core of the design is made up of four elements. 
+`spec.feature_a` is the name of the feature that is being configured in that section.
+`spec.feature_a.configure` is a boolean field which tells the kuadrant operator to use the configuration defined in the Kuadrant CR for the feature. 
+This is done to allow us the flexibility to have features active by default.
+`spec.feature_a.enable` is a list of strings that represent the components the configuration will be applied too.
+Having it a list of strings allows us to extend or reduce the number of accepted values without needing to modify the CRD. 
+With the use of strings as values means keywords can also be defined, for example `"default"` could be shorthand for a sub set of components affected by the feature.
+`spec.feature_a.disable` is a list of strings that repenting the components that the configuration will *not* be applied too.
```

Comment:
I think this could be added at a later date if it was shown to be needed. We don't have that many components and so an enable list should not be too onerous on the user IMO

Reply:
You maybe right, but I would like to have a guide to what happens if the disable is required. I don't like the idea of updating the CRD with new fields in the future, if we can plan for it now.